emu/src/cpu/arm/alu_instruction.rs

```diff
@@ -219,23 +220,66 @@ impl From<u32> for PsrOpKind {
             && op_code.get_bits(16..=21) == 0b001111
             && op_code.get_bits(0..=11) == 0b0000_0000_0000
         {
-            Self::Mrs
+            Self::Mrs {
+                destination_register: op_code.get_bits(12..=15),
+            }
         } else if op_code.get_bits(23..=27) == 0b00010
             && op_code.get_bits(12..=21) == 0b10_1001_1111
             && op_code.get_bits(4..=11) == 0b0000_0000
         {
-            Self::Msr
+            Self::Msr {
+                source_register: op_code.get_bits(0..=3),
+            }
         } else if op_code.get_bits(26..=27) == 0b00
             && op_code.get_bits(23..=24) == 0b10
             && op_code.get_bits(12..=21) == 0b10_1000_1111
         {
-            Self::MsrFlg
+            Self::MsrFlg {
+                operand: if op_code.get_bit(25) {
+                    AluSecondOperandInfo::Immediate {
+                        base: op_code.get_bits(0..=7),
+                        shift: op_code.get_bits(8..=11) * 2,
```

Comment:
`* 2`? We usually use << N

Reply:
I used the same logic as in `emu/src/cpu/arm/instructions.rs` line `472`